Classes/TextUtils.cs

```diff
@@ -92,6 +98,37 @@ public static string StripDateFromName(string name, DateTime stripDate)
             return name;
         }
 
+        /// <summary>
+        /// Derive ‘SmartName’ from the ProgrammeName and EpisodeName values
+        /// Use regex to remove a number of different any duplication of the Programme Name.
+        /// </summary>
+        /// <param name="programmeName">Programme Name to be checked.</param>
+        /// <param name="episodeName">Episode Name to be used to form SmartName.</param>
+        /// <returns>Episode name with date content removed</returns>
+        public static string StripProgrammeNameFromEpisode(string programmeName, string episodeName)
+        {
+            string smartname = string.Empty;
+
+            // Construct smanrtName from ProgrammeName and EpisodeName
+            smartname = programmeName + ": " + episodeName;
```

Comment:
I'd suggest removing the `smartname` variable entirely and then doing a `return programmeName + ": " + episodeName;` at the end of the function to avoid building the smart name twice.

Reply:
okay, that makes sense. will change this.